src/components/CodeManagement/CodeManagementRoutes.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  useHistory,
+  useRouteMatch,
+  Switch,
+  Route,
+} from 'react-router-dom';
+import { Tabs, Tab } from '@edx/paragon';
+
+import ManageCodesTab from './ManageCodesTab';
+import ManageRequestsTab from './ManageRequestsTab';
+import { ROUTE_NAMES } from '../EnterpriseApp/constants';
+import { features } from '../../config';
+
+const CodeManagementRoutes = ({ enterpriseSlug }) => {
+  const isTabsFeatureEnabled = !!features.FEATURE_BROWSE_AND_REQUEST;
+
+  const routesByTabKey = {
+    requests: `/${enterpriseSlug}/admin/${ROUTE_NAMES.codeManagement}/requests`,
+    default: `/${enterpriseSlug}/admin/${ROUTE_NAMES.codeManagement}`,
+  };
+  const history = useHistory();
+  const requestsTabMatch = useRouteMatch(routesByTabKey.requests);
+  const initialTabKey = requestsTabMatch ? 'requests' : 'default';
+  const [tabKey, setTabKey] = useState(initialTabKey);
+
+  useEffect(() => {
+    if (!isTabsFeatureEnabled) {
+      return;
+    }
+
+    if (requestsTabMatch) {
+      setTabKey('requests');
+    } else {
+      setTabKey('default');
+    }
+  }, [isTabsFeatureEnabled, requestsTabMatch]);
+
+  const handleTabSelect = (key) => {
+    if (key === 'requests') {
+      history.push(routesByTabKey.requests);
+    } else if (key === 'default') {
+      history.push(routesByTabKey.default);
+    }
+    setTabKey(key);
+  };
+
+  if (isTabsFeatureEnabled) {
+    return (
+      <Switch>
+        <Tabs
+          id="tabs-code-management"
+          activeKey={tabKey}
```

Comment:
I would consider having a single source of truth for the state of `activeKey`. I would recommend not using `useState` but instead using `useParams`. In [the settings page](https://github.com/openedx/frontend-app-admin-portal/blob/7a8a3174911f6b4b2d0e75e1478c149593cd125e/src/components/settings/SettingsTabs.jsx#L25) we actually have a simple custom hook that does this for us.

I think doing something like that will make this simpler and remove the need to use `useEffect`.

Reply:
@manny-m Good suggestion, updated. Originally didn't go with this approach because the existing routing structure of subscription management wasn't quite the same as the settings page, and I was originally trying to avoid creating a new URL (e.g., /subscriptions -> /subscriptions/manage-learners).

That said, I agree with your feedback and perhaps this is a decision we should document for this repo. That is, when implementing tabs, each tab should have its own key param for a route so we're consistent with tabs throughout the application. ADR worthy?